django/db/migrations/serializer.py

```diff
@@ -0,0 +1,387 @@
+from __future__ import unicode_literals
+
+import collections
+import datetime
+import decimal
+import functools
+import math
+import types
+from importlib import import_module
+
+from django.db import models
+from django.db.migrations.operations.base import Operation
+from django.db.migrations.utils import COMPILED_REGEX_TYPE, RegexObject
+from django.utils import datetime_safe, six
+from django.utils.encoding import force_text
+from django.utils.functional import LazyObject, Promise
+from django.utils.timezone import utc
+from django.utils.version import get_docs_version
+
+try:
+    import enum
+except ImportError:
+    # No support on Python 2 if enum34 isn't installed.
+    enum = None
+
+
+class BaseSerializer(object):
+    def __init__(self, value):
+        self.value = value
+
+    def serialize(self):
+        raise NotImplementedError
+
+
+class SimpleSerializer(BaseSerializer):
+    def serialize(self):
+        return repr(self.value), set()
+
+
+class FloatSerializer(SimpleSerializer):
+    def serialize(self):
+        if math.isnan(self.value) or math.isinf(self.value):
+            return 'float("{}")'.format(self.value), set()
+        return super(FloatSerializer, self).serialize()
+
+
+class SequenceSerializer(BaseSerializer):
```

Comment:
Rename to `BaseSequenceSerializer`, make the `_format()` raise a `NotImplementedError` similar to the `BaseSerializer`. Then add a `ListSerializer` along `TupleSerializer` etc. that implements the `_format()` method.

```python
class BaseSequenceSerializer(BaseSerializer):
    def _format(self):
        raise ...

class ListSerializer(BaseSequenceSerializer):
    def _format(self):
        return "[%s]"

class TupleSerializer(BaseSequenceSerializer):
    # as already implemented
```